docs/guides/03_tracking_qubo.ipynb

```diff
@@ -0,0 +1,253 @@
+   "id": "2998ef49",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Problem name: qubo_trial\n",
+      "\n",
+      "Minimize\n",
+      "  x*y - x*z + 2*y*z + x - 2*y + 3*z\n",
+      "\n",
+      "Subject to\n",
+      "  No constraints\n",
+      "\n",
+      "  Binary variables (3)\n",
+      "    x y z\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "qubo = QuadraticProgram(name=\"qubo_trial\")\n",
+    "qubo.binary_var(\"x\")\n",
+    "qubo.binary_var(\"y\")\n",
+    "qubo.binary_var(\"z\")\n",
+    "qubo.minimize(linear=[1, -2, 3], quadratic={(\"x\", \"y\"): 1, (\"x\", \"z\"): -1, (\"y\", \"z\"): 2})\n",
+    "print(qubo.prettyprint())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "503f1e7e",
+   "metadata": {},
+   "source": [
+    "## QAOA"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f7ca3467",
+   "metadata": {},
+   "source": [
+    "We setup our experiment using QAOA."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "d30da06f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "qaoa_mes = QAOA(sampler=Sampler(), optimizer=COBYLA())\n",
+    "qaoa = MinimumEigenOptimizer(qaoa_mes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "856bcbd0",
+   "metadata": {},
+   "source": [
+    "Now let's register this experiment in `LocalStorage`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "f8b20245",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "local_storage = LocalStorage(\"./trials\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "id": "17f03646",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with Trial(\"QAOA trial\", storage=local_storage) as trial:\n",
+    "    # track some parameters\n",
+    "    trial.add_parameter(\"algo\", \"qaoa\")\n",
+    "    trial.add_parameter(\"sampler\", \"qiskit.primitives.Sampler\")\n",
+    "    trial.add_parameter(\"optimizer\", \"qiskit.algorithms.optimizers.COBYLA\")\n",
+    "    \n",
+    "    # track usefull data\n",
+    "    trial.add_text(\"qubo\", qubo.export_as_lp_string())\n",
+    "    \n",
+    "    # run\n",
+    "    qaoa_result = qaoa.solve(qubo)\n",
+    "    \n",
+    "    # track results of run\n",
+    "    trial.add_text(\"qaoa_result\", qaoa_result.prettyprint())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "370bf837",
+   "metadata": {},
+   "source": [
+    "## VQE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "53fe53ae",
+   "metadata": {},
+   "source": [
+    "We setup our experiment using VQE."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "id": "bebfca16",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ansatz = TwoLocal(rotation_blocks=\"ry\", entanglement_blocks=\"cz\")\n",
+    "vqe_mes = SamplingVQE(sampler=Sampler(), ansatz=ansatz, optimizer=COBYLA())\n",
```

Comment:
I think VQE has callback parameter. Can we move vqe initialization inside trial and log histories too?

I do not remember syntax exactly, but it will be something like that:
```python
with Trial("...") as trial:
    vqe = VQE(..., callback=lambda idx, params, mean, std: trial.add_metric("vqe_history", mean))
```